lib/dry/inflector/inflections.rb

```diff
@@ -69,6 +73,8 @@ def initialize
         @singulars    = Rules.new
         @humans       = Rules.new
         @uncountables = Set[]
+        @acronyms     = Hash.new
+        define_acronym_regex_patterns
```

Comment:
@GustavoCaso IMO this is a symptom that `@acronyms` should be something else than a `Hash`. This because the "holder" object `Inflections` is "worried" about the state of `@acronyms`, so it has to instantiate a data structure and then constantly maintain it via `#define_acronym_regex_patterns`.

What if we try first with `Rules`, to check if that object can serve the purpose. Otherwise we can define `Acronyms` object that just wraps the `Hash` and exposes one method to add an entry and one method to match the internal regexps?

Reply:
Makes total sense @jodosha I will try to experiment with it when I find some time, I was trying to make this issue rolling so the more people contribute the better. Thanks @flash-gordon @jodosha @AMHOL